5.0 Draft/IDD/IDDs Orchestration System/eu.arrowhead.orchestration-pull-http-json.yml

```diff
@@ -0,0 +1,303 @@
+    uses the orchestration store policies to recommend service providers to the requesting consumer to fulfil 
+    its task. 
+
+    The consumer system request for orchestration either by specifying the service definition that the consumer requires or 
+    by using stored SoS configurations or Orchestration rules if no service information is defined. 
+
+    _SoS configurations_ represents the connections needed in a SoS for it to perform its designed task. This means, describing what services
+    must each of the systems consume to perform their individual tasks. A SoS configuration can be seen as a series of _Orchestration rules_.
+    _Orchestration rules_ represents a desired interaction between systems consuming specific services from specific preferred providers. 
+
+    In the context of this document, we reffer to the Orchestration consumer as _requesting-system_ where _system-id_ is the
+    unique identifier that represents that system. The _requested-service_ represents the service information that the _requesting-system_
+    requires orchestration. In case the _requested_service_ is specified, the requires service definition must be included in the form, 
+    in addition, at least one interface must be specified. The orchestration system will attempt to match the _requested-service_ information
+    with services present on the orchestration rules or SoS configurations. 
+
+version: 5.0.0-rc.1
+
+# Additional information present in the IDD template
+x-transfer-protocol: [HTTP, HTTPS]
+x-encryption-method: to-add
+x-encoding: JSON
+x-compression: to-add
+x-semantics: to-add
+x-Ontology: to-add
+
+paths:
+  /orchestration/echo:
+    get:  
+      summary: Endpoint to check the core system availability. 
+      description: Returns an OK to indicate that the core system is available. 
+      responses:
+        '200':
+          description: OK
+
+
+  /orchestration/orchestration-pull:
+    post:
+      summary: Submit a service request and get orchestration response. If no service definition is specified, the orchestration
+        will return the suggested services and their appropriate providers for the consuming system to fulfil its tasks, depending
+        on a existing SoS configuration or Orchestration rules. 
+
+        Information of the requesting system is mandatory. Requested service is optional. If a requested service object is specified,
+        the service definition and at least one interface must be specified. 
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ServiceRequestForm'
+      responses:
+        '200':
+          description: Successful orchestration response.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/OrchestrationResponse'
+
+  /orchestration/orchestration-pull/{id}:
```

Comment:
If I can have all my wishes come true ;), then I'd have only one endpoint for requesting service orchestration. Can we get away with removing this one?

Reply:
I'm ambivalent to removing the pull service. My reluctance is related to the producer/consumer situation with orchestration.  I can definitely see the benefit of having less number of services, in particular for the Core Services. 

But, in order for the orchestration system to serve configurations to systems, I would prefer a producer service at the orchestration system. This service should be consumed by the application system at the systems initiative. In this way the application do not have to wait for orchestration to take place and ultimately be dependent of the alertness of the orchestration system.

This service originally was named "orchestration store" - meaning that it was the storage of service configurations. This implicates that the pull mechanism was the primary means of acquire service configurations.